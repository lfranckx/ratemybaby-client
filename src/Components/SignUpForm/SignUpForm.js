import React, { Component } from 'react';
import AuthApiService from '../../Services/auth-api-service'
import BabyContext from '../../Contexts/BabyContext'

export default class SignUp extends Component {
    static defaultProps = {
        onRegistrationSuccess: () => {}
    }

    static contextType = BabyContext

    state = { error: null }

    handleSubmit = ev => {
        ev.preventDefault()
        const { username, password, email } = ev.target

        this.setState({ error: null })
        AuthApiService.postUser({
            username: username.value,
            user_password: password.value,
            email: email.value,
            baby_ids: []
        })
        .then(res => {
            AuthApiService.postLogin({
                username: username.value,
                user_password: password.value
            })
            .then(user => {
                username.value = ''
                password.value = ''
                console.log('SignUpForm getUser response:', user);
                this.props.onSignUpSuccess()
            })
        })
        .catch(res => {
            this.setState({ error: res.error })
        })
    }

    render() {
        const { error } = this.state
        return (
            <form 
                id='sign-up-form'
                onSubmit={this.handleSubmit}
            >
                <div role='alert'>
                    {error && <p className='error'>{error}</p>}
                </div>
                <div className="input-box">
                    <label className='label'>
                        Email Address
                        <span className='required'>*</span>
                    </label>
                    <input 
                        type="text" 
                        name='email' 
                        required
                        id='email' />
                </div>
                <div className="input-box">
                    <label className='label'>
                        Username
                        <span className='required'>*</span>
                    </label>
                    <input 
                        type="text" 
                        name='username' 
                        required
                        className='username' />
                </div>
                <div className="input-box">
                    <label className='label'>
                        Password
                        <span className='required'>*</span>
                    </label>                    
                    <input 
                        type="password" 
                        name='password'
                        required
                        className='password' />
                </div>
                <button id="sign-up-button" type="submit">Sign Up</button>
            </form>
        )
    }
}