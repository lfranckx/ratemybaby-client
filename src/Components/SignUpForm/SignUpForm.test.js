import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignUpForm from './SignUpForm';

const simulateChangeOnInput = (wrapper, inputSelector, newValue) => {
    let input = wrapper.find(inputSelector);
    input.simulate('change', {
        target: { value: newValue }
    });
    return wrapper.find(inputSelector);
};

describe(`SignUpForm component`, () => {
    const props = {
        handleSignUpSuccess: () => {}
    };

    it(`renders form given props`, () => {
        const wrapper = shallow(<SignUpForm {...props}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it(`lets me fill out the form and submit it`, () => {
        const wrapper = shallow(<SignUpForm {...props}/>);
        const emailInput = simulateChangeOnInput(wrapper, '#signinemail', 'test@email.com');
        const userInput = simulateChangeOnInput(wrapper, '#signinuser', 'test');
        const pwInput = simulateChangeOnInput(wrapper, '#signinpass', 'Test1234!');

        console.log('emailInput', emailInput, 'userInput', userInput)

        expect(emailInput.props().value).toEqual('test@email.com');
        expect(userInput.props().value).toEqual('test');
        expect(pwInput.props().value).toEqual('Test1234!');

        const submitButton = wrapper.find('#sign-up-button');
        submitButton.simulate('click')
    });
});