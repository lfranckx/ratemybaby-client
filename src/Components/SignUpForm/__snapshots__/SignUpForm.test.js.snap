// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm component renders form given props 1`] = `
<form
  id="sign-up-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="input-box"
  >
    <label
      className="label"
    >
      Email Address
      <span
        className="required"
      >
        *
      </span>
    </label>
    <input
      aria-label="email"
      id="signinemail"
      name="email"
      required={true}
      type="text"
    />
  </div>
  <div
    className="input-box"
  >
    <label
      className="label"
    >
      Username
      <span
        className="required"
      >
        *
      </span>
    </label>
    <input
      aria-label="username"
      className="username"
      id="signinuser"
      name="username"
      required={true}
      type="text"
    />
  </div>
  <div
    className="input-box"
  >
    <label
      className="label"
    >
      Password
      <span
        className="required"
      >
        *
      </span>
    </label>
    <input
      aria-label="password"
      autoComplete="on"
      className="password"
      id="signinpass"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    id="sign-up-button"
    type="submit"
  >
    Sign Up
  </button>
</form>
`;
